tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_cell_data.py

```diff
@@ -26,16 +26,10 @@ def test_enable_cell_data_via_settings_app(self):
 
         # deal with prompt that sometimes appears (on first setting)
         if cell_data_prompt.is_displayed:
+            self.assertTrue(cell_and_data_settings.is_data_toggle_checked)
```

Comment:
The data connection checkbox is checked at this point, but data connection is only enabled after tapping "Confirm" in the confirmation dialog.
If you tap "Not now" in the confirmation dialog, you see that the data connection checkbox is turned off again (we should perhaps add a test for this?).
I'll add a comment that explains this.